### DISCUSSION TITLE
Build  zephyr  application ‘micropython’ for (QEMU)qemu_x86 failed.

### DISCUSSION THREAD

**Discussion by stc2001** [zephyrproject-rtos/zephyr#35807] [Q&A]
👀
Log:
gexingdeyun@gexingdeyun-ThinkPad-L440:~/zephyrproject$ west build -p -b qemu_x86  ~/micropython/ports/zephyr
-- west build: making build dir /home/gexingdeyun/zephyrproject/build pristine
-- west build: generating a build system
Including boilerplate (Zephyr base): /home/gexingdeyun/zephyrproject/zephyr/cmake/app/boilerplate.cmake
-- Application: /home/gexingdeyun/micropython/ports/zephyr
-- Zephyr version: 2.6.0-rc2 (/home/gexingdeyun/zephyrproject/zephyr), build: v2.6.0-rc2-39-g94e8c2eb81f3
-- Found Python3: /usr/bin/python3.9 (found suitable exact version "3.9.5") found components: Interpreter 
-- Found west (found suitable version "0.11.0", minimum required is "0.7.1")
-- Board: qemu_x86
-- Cache files will be written to: /home/gexingdeyun/.cache/zephyr
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Using toolchain: zephyr 0.12.4 (/home/gexingdeyun/zephyr-sdk-0.12.4)
-- Found dtc: /home/gexingdeyun/zephyr-sdk-0.12.4/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.5.0", minimum required is "1.4.6")
-- Found BOARD.dts: /home/gexingdeyun/zephyrproject/zephyr/boards/x86/qemu_x86/qemu_x86.dts
-- Generated zephyr.dts: /home/gexingdeyun/zephyrproject/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /home/gexingdeyun/zephyrproject/build/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: /home/gexingdeyun/zephyrproject/build/zephyr/include/generated/device_extern.h

warning: QEMU_ICOUNT_SHIFT (defined at boards/Kconfig:40) was assigned the value '5' but got the
value ''. Check these unsatisfied dependencies: QEMU_ICOUNT (=n). See
http://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_QEMU_ICOUNT_SHIFT.html and/or look up
QEMU_ICOUNT_SHIFT in the menuconfig/guiconfig interface. The Application Development Primer, Setting
Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be helpful
too.

warning: CONSOLE_GETCHAR_BUFSIZE (defined at subsys/console/Kconfig:29) was assigned the value '128'
but got the value ''. Check these unsatisfied dependencies: CONSOLE_GETCHAR (=n), CONSOLE_SUBSYS
(=n). See http://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_CONSOLE_GETCHAR_BUFSIZE.html
and/or look up CONSOLE_GETCHAR_BUFSIZE in the menuconfig/guiconfig interface. The Application
Development Primer, Setting Configuration Values, and Kconfig - Tips and Best Practices sections of
the manual might be helpful too.

warning: CONSOLE_PUTCHAR_BUFSIZE (defined at subsys/console/Kconfig:38) was assigned the value '128'
but got the value ''. Check these unsatisfied dependencies: CONSOLE_GETCHAR (=n), CONSOLE_SUBSYS
(=n). See http://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_CONSOLE_PUTCHAR_BUFSIZE.html
and/or look up CONSOLE_PUTCHAR_BUFSIZE in the menuconfig/guiconfig interface. The Application
Development Primer, Setting Configuration Values, and Kconfig - Tips and Best Practices sections of
the manual might be helpful too.

warning: The choice symbol CONSOLE_GETCHAR (defined at subsys/console/Kconfig:16) was selected (set
=y), but no symbol ended up as the choice selection. See
http://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_CONSOLE_GETCHAR.html and/or look up
CONSOLE_GETCHAR in the menuconfig/guiconfig interface. The Application Development Primer, Setting
Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be helpful
too.

Parsing /home/gexingdeyun/micropython/ports/zephyr/Kconfig
Loaded configuration '/home/gexingdeyun/zephyrproject/zephyr/boards/x86/qemu_x86/qemu_x86_defconfig'
Merged configuration '/home/gexingdeyun/micropython/ports/zephyr/prj.conf'
Merged configuration '/home/gexingdeyun/micropython/ports/zephyr/boards/qemu_x86.conf'
Configuration saved to '/home/gexingdeyun/zephyrproject/build/zephyr/.config'
Kconfig header saved to '/home/gexingdeyun/zephyrproject/build/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 10.2.0
-- The CXX compiler identification is GNU 10.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /home/gexingdeyun/zephyr-sdk-0.12.4/x86_64-zephyr-elf/bin/x86_64-zephyr-elf-gcc
CMake Warning at /home/gexingdeyun/zephyrproject/zephyr/subsys/random/CMakeLists.txt:11 (message):
  

      Warning: CONFIG_TEST_RANDOM_GENERATOR is not a truly random generator.
      This capability is not secure and it is provided for testing purposes only.
      Use it carefully.

-- Found Python3: /usr/bin/python3.9 (found version "3.9.5") found components: Interpreter 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/gexingdeyun/zephyrproject/build
-- west build: building application
[1/432] Preparing syscall dependency handling

[10/432] Generating genhdr/mpversion.h
GEN /home/gexingdeyun/zephyrproject/build/genhdr/mpversion.h
[99/432] Generating genhdr/qstrdefs.collected.h
QSTR updated
[246/432] Building C object CMakeFiles...un/micropython/extmod/modonewire.c.obj
In file included from /home/gexingdeyun/micropython/ports/zephyr/../../py/mphal.h:35,
                 from /home/gexingdeyun/micropython/extmod/modonewire.c:31:
/home/gexingdeyun/micropython/extmod/modonewire.c: In function 'onewire_bus_reset':
/home/gexingdeyun/micropython/ports/zephyr/mphalport.h:40:37: warning: implicit declaration of function 'mp_raise_NotImplementedError' [-Wimplicit-function-declaration]
   40 | #define mp_hal_pin_od_low(p)       (mp_raise_NotImplementedError("mp_hal_pin_od_low"))
      |                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/gexingdeyun/micropython/extmod/modonewire.c:47:5: note: in expansion of macro 'mp_hal_pin_od_low'
   47 |     mp_hal_pin_od_low(pin);
      |     ^~~~~~~~~~~~~~~~~
/home/gexingdeyun/micropython/extmod/modonewire.c:49:18: warning: implicit declaration of function 'mp_hal_quiet_timing_enter' [-Wimplicit-function-declaration]
   49 |     uint32_t i = mp_hal_quiet_timing_enter();
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/gexingdeyun/micropython/extmod/modonewire.c:53:5: warning: implicit declaration of function 'mp_hal_quiet_timing_exit' [-Wimplicit-function-declaration]
   53 |     mp_hal_quiet_timing_exit(i);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~
[289/432] Building C object CMakeFiles/micropython.dir/modusocket.c.obj
FAILED: CMakeFiles/micropython.dir/modusocket.c.obj 
ccache /home/gexingdeyun/zephyr-sdk-0.12.4/x86_64-zephyr-elf/bin/x86_64-zephyr-elf-gcc -DBUILD_VERSION=v2.6.0-rc2-39-g94e8c2eb81f3 -DFFCONF_H=\"/home/gexingdeyun/micropython/ports/zephyr/../../lib/oofatfs/ffconf.h\" -DKERNEL -DMBEDTLS_CONFIG_FILE=\"config-tls-generic.h\" -DMICROPY_HEAP_SIZE=16384 -DMICROPY_VFS_FAT=1 -DMICROPY_VFS_LFS1=0 -DMICROPY_VFS_LFS2=1 -DMP_CONFIGFILE="<mpconfigport.h>" -DNDEBUG -D_FORTIFY_SOURCE=2 -D__LINUX_ERRNO_EXTENSIONS__ -D__ZEPHYR__=1 -I/home/gexingdeyun/micropython/ports/zephyr/../.. -I/home/gexingdeyun/micropython/ports/zephyr -I. -I/home/gexingdeyun/zephyrproject/zephyr/include -Izephyr/include/generated -I/home/gexingdeyun/zephyrproject/zephyr/soc/x86/ia32 -I/home/gexingdeyun/zephyrproject/zephyr/lib/libc/newlib/include -I/home/gexingdeyun/zephyrproject/zephyr/subsys/net/lib/sockets/. -I/home/gexingdeyun/zephyrproject/zephyr/subsys/net/lib/dns/. -I/home/gexingdeyun/zephyrproject/zephyr/subsys/net/l2 -I/home/gexingdeyun/zephyrproject/modules/crypto/mbedtls/mbedtls/include -I/home/gexingdeyun/zephyrproject/zephyr/modules/mbedtls/configs -m32  -Os -imacros /home/gexingdeyun/zephyrproject/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -fdiagnostics-color=always -m32 -Wa,--divide -imacros /home/gexingdeyun/zephyrproject/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-address-of-packed-member -Wno-unused-but-set-variable -Werror=implicit-int -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/home/gexingdeyun/micropython/ports/zephyr=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/gexingdeyun/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/gexingdeyun/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections -mpreferred-stack-boundary=2 -mno-mmx -mno-sse -std=c99 -std=gnu99 -fomit-frame-pointer -MD -MT CMakeFiles/micropython.dir/modusocket.c.obj -MF CMakeFiles/micropython.dir/modusocket.c.obj.d -o CMakeFiles/micropython.dir/modusocket.c.obj -c /home/gexingdeyun/micropython/ports/zephyr/modusocket.c
In file included from /home/gexingdeyun/zephyrproject/zephyr/include/net/socket.h:29,
                 from /home/gexingdeyun/micropython/ports/zephyr/modusocket.c:41:
/home/gexingdeyun/zephyrproject/zephyr/include/net/socket_select.h:108: warning: "fd_set" redefined
  108 | #define fd_set zsock_fd_set
      | 
In file included from /home/gexingdeyun/zephyr-sdk-0.12.4/x86_64-zephyr-elf/x86_64-zephyr-elf/sys-include/sys/types.h:50,
                 from /home/gexingdeyun/zephyr-sdk-0.12.4/x86_64-zephyr-elf/x86_64-zephyr-elf/sys-include/stdio.h:61,
                 from /home/gexingdeyun/micropython/ports/zephyr/modusocket.c:33:
/home/gexingdeyun/zephyr-sdk-0.12.4/x86_64-zephyr-elf/x86_64-zephyr-elf/sys-include/sys/select.h:46: note: this is the location of the previous definition
   46 | #define fd_set _types_fd_set
      | 
In file included from /home/gexingdeyun/zephyrproject/zephyr/include/net/socket.h:29,
                 from /home/gexingdeyun/micropython/ports/zephyr/modusocket.c:41:
/home/gexingdeyun/zephyrproject/zephyr/include/net/socket_select.h:110: warning: "FD_SETSIZE" redefined
  110 | #define FD_SETSIZE ZSOCK_FD_SETSIZE
      | 
In file included from /home/gexingdeyun/zephyr-sdk-0.12.4/x86_64-zephyr-elf/x86_64-zephyr-elf/sys-include/sys/types.h:50,
                 from /home/gexingdeyun/zephyr-sdk-0.12.4/x86_64-zephyr-elf/x86_64-zephyr-elf/sys-include/stdio.h:61,
                 from /home/gexingdeyun/micropython/ports/zephyr/modusocket.c:33:
/home/gexingdeyun/zephyr-sdk-0.12.4/x86_64-zephyr-elf/x86_64-zephyr-elf/sys-include/sys/select.h:31: note: this is the location of the previous definition
   31 | # define FD_SETSIZE 64
      | 
In file included from /home/gexingdeyun/zephyrproject/zephyr/include/net/socket.h:29,
                 from /home/gexingdeyun/micropython/ports/zephyr/modusocket.c:41:
/home/gexingdeyun/zephyrproject/zephyr/include/net/socket_select.h:112:19: error: conflicting types for 'select'
  112 | static inline int select(int nfds, zsock_fd_set *readfds,
      |                   ^~~~~~
In file included from /home/gexingdeyun/zephyr-sdk-0.12.4/x86_64-zephyr-elf/x86_64-zephyr-elf/sys-include/sys/types.h:50,
                 from /home/gexingdeyun/zephyr-sdk-0.12.4/x86_64-zephyr-elf/x86_64-zephyr-elf/sys-include/stdio.h:61,
                 from /home/gexingdeyun/micropython/ports/zephyr/modusocket.c:33:
/home/gexingdeyun/zephyr-sdk-0.12.4/x86_64-zephyr-elf/x86_64-zephyr-elf/sys-include/sys/select.h:62:5: note: previous declaration of 'select' was here
   62 | int select __P ((int __n, fd_set *__readfds, fd_set *__writefds,
      |     ^~~~~~
/home/gexingdeyun/zephyrproject/zephyr/include/net/socket_select.h:119:20: error: expected identifier or '(' before '__extension__'
  119 | static inline void FD_ZERO(zsock_fd_set *set)
      |                    ^~~~~~~
/home/gexingdeyun/zephyrproject/zephyr/include/net/socket_select.h:124:49: error: expected ')' before '*' token
  124 | static inline int FD_ISSET(int fd, zsock_fd_set *set)
      |                                                 ^
/home/gexingdeyun/zephyrproject/zephyr/include/net/socket_select.h:124:19: error: expected ')' before '->' token
  124 | static inline int FD_ISSET(int fd, zsock_fd_set *set)
      |                   ^~~~~~~~
/home/gexingdeyun/zephyrproject/zephyr/include/net/socket_select.h:129:48: error: expected ')' before '*' token
  129 | static inline void FD_CLR(int fd, zsock_fd_set *set)
      |                                                ^
/home/gexingdeyun/zephyrproject/zephyr/include/net/socket_select.h:129:20: error: expected ')' before '->' token
  129 | static inline void FD_CLR(int fd, zsock_fd_set *set)
      |                    ^~~~~~
/home/gexingdeyun/zephyrproject/zephyr/include/net/socket_select.h:134:48: error: expected ')' before '*' token
  134 | static inline void FD_SET(int fd, zsock_fd_set *set)
      |                                                ^
/home/gexingdeyun/zephyrproject/zephyr/include/net/socket_select.h:134:20: error: expected ')' before '->' token
  134 | static inline void FD_SET(int fd, zsock_fd_set *set)
      |                    ^~~~~~
[298/432] Building C object CMakeFiles...un/micropython/lib/littlefs/lfs2.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /home/gexingdeyun/zephyrproject/build

**Comment by MaureenHelm** [comment]
Please see micropython/micropython#7292 for a fix
